marimo/_server/export/exporter.py

```diff
@@ -118,16 +121,41 @@ def export_as_script(
             graph.cells[cid].code
             for cid in dataflow.topological_sort(graph, graph.cells.keys())
         ]
-        filename = file_manager.filename
-        if not filename:
-            filename = "notebook.script.py"
-
-        basename = os.path.basename(filename)
-        download_filename = f"{os.path.splitext(basename)[0]}.script.py"
-
         code = (
             f'\n__generated_with = "{__version__}"\n\n'
             + "\n\n# ---\n\n".join(codes)
         )
 
+        download_filename = get_download_filename(file_manager, ".script.py")
         return code, download_filename
+
+    def export_as_md(self, file_manager: AppFileManager) -> tuple[str, str]:
+        # TODO: Provide filter or kernel in header such that markdown documents
+        # are executable.
+        document = [
+            dedent(
+                f"""
+          ---
+          title: {get_filename_title(file_manager)}
```

Comment:
just recently (after this PR was put up), someone added `app_title`, which you could use as the title. it should be under `app._config.app_title`, and then can callback to `get_filename_title()`